d2/src/main/java/com/linkedin/d2/xds/balancer/DualReadXdsLoadBalancerFactory.java

```diff
@@ -0,0 +1,31 @@
+package com.linkedin.d2.xds.balancer;
+
+import com.linkedin.d2.balancer.D2ClientConfig;
+import com.linkedin.d2.balancer.LoadBalancerWithFacilities;
+import com.linkedin.d2.balancer.LoadBalancerWithFacilitiesFactory;
+import com.linkedin.d2.balancer.dualread.DualReadLoadBalancer;
+import com.linkedin.d2.balancer.dualread.DualReadModeProvider;
+import java.util.concurrent.Executors;
+
+
+public class DualReadXdsLoadBalancerFactory implements LoadBalancerWithFacilitiesFactory
```

Comment:
If we want to make this factory general for any dual read situation in the future, we should remove the "Xds" part. However, I don't think we will have more dual read cases since restli clients should be deprecated by gRPC clients in near future. I think we can just name it particularly as "ZkAndXds", and name the new and old factory directly as xdsLbFactory and zkLbFactory.